lib/util/http/gerrit.ts

```diff
@@ -0,0 +1,48 @@
+import is from '@sindresorhus/is';
+import JSON5 from 'json5';
+import type { HttpOptions, HttpResponse, InternalHttpOptions } from './types';
+import { Http } from './index';
+
+let baseUrl: string;
+export const setBaseUrl = (url: string): void => {
+  baseUrl = url;
+};
+
+/**
+ * Access Gerrit REST-API and strip-of the "magic prefix" from responses.
+ * @see https://gerrit-review.googlesource.com/Documentation/rest-api.html
+ */
+export class GerritHttp extends Http {
+  magicPrefix = /^\)]}'\n/g;
+
+  constructor(options?: HttpOptions) {
+    super('gerrit', options);
+  }
+
+  protected override async request<T>(
+    path: string,
+    options?: InternalHttpOptions
+  ): Promise<HttpResponse<T>> {
+    const url = baseUrl + path;
```

Comment:
use joinUrlParts

Reply:
this will not work. The `joinUrlParts` [will strip the trailing slashes](https://github.com/jfromaniello/url-join/issues/10)...

Reply:
Why do we need the trailing slashes?

I think we need to use bare `path` if it's already a full url like we do in other implementations.

Reply:
> I think we need to use bare `path` if it's already a full url like we do in other implementations.

`path` is not the full url.

Reply:
are we still talking about `joinUrlParts`?

You want to support to use the `GerritHttp` class for other purposes and without setting `setBaseUrl` previously?

i can check if baseUrl is undefined like
```
let baseUrl: string|undefined;
...
const url = baseUrl ? baseUrl + path : path;
```

or even more complex checks if `path` is already an url...

Reply:
check if path is already an absolute path and then skip the baseUrl prefix if already absolute